libraries/node-core-library/src/PackageJsonLookup.ts

```diff
@@ -292,10 +292,18 @@ export class PackageJsonLookup {
       return this._packageFolderCache.get(resolvedFileOrFolderPath);
     }
 
-    // Is resolvedFileOrFolderPath itself a folder with a package.json file?  If so, return it.
-    if (FileSystem.exists(path.join(resolvedFileOrFolderPath, FileConstants.PackageJson))) {
-      this._packageFolderCache.set(resolvedFileOrFolderPath, resolvedFileOrFolderPath);
-      return resolvedFileOrFolderPath;
+    // Is resolvedFileOrFolderPath itself a folder with a valid package.json file?  If so, return it.
+    const packageJsonFilePath: string = path.join(resolvedFileOrFolderPath, FileConstants.PackageJson);
+    if (FileSystem.exists(packageJsonFilePath)) {
+      try {
+        this.loadNodePackageJson(packageJsonFilePath);
+        this._packageFolderCache.set(resolvedFileOrFolderPath, resolvedFileOrFolderPath);
+        return resolvedFileOrFolderPath;
+      } catch (error) {
+        if (!/The required field "name" was not found/.test(error.message)) {
```

Comment:
Instead of testing for the prose in an error message, can this set a well-known constant in the `error.code` field and check for that?

Reply:
That's certainly possible. Is there prior art of a `rushstack` custom error doing so, or should I figure out a decent pattern for that independently?

Reply:
Could we solve it without throwing an exception at all?  Throwing exceptions around during successful operations can hurt performance and it also interferes with debugging.

Reply:
Yes. As one possible non-breaking fix, `loadNodePackageJson(jsonFilename: string)` could get a second optional boolean argument which would allow it to return `null` if the `"name"` field is not present, and this could be detected in `_tryGetPackageFolderFor(resolvedFileOrFolderPath: string)`.

Should I add that change to this PR?

Reply:
`loadNodePackageJson()` is so-named because it reports an error if it cannot load the file.  It would be called `tryLoadNodePackageJson()` if it could return `undefined`.  It might be better to introduce private functions to avoid changing the public API signature.

I don't have time right now to propose a more specific solution, but let us know if you need us to recommend something.

Reply:
Thank you @octogonz, that was (hopefully!) sufficient guidance for me to figure out the shape of the fix that you might be looking for. Updated the PR accordingly.

Reply:
@iclanton @octogonz  this was moved to "Waiting for Author", but has since been updated. Doesn't seem there's any way to move it out of "Waiting for Author"